resources/metadata/codeqlExecuteScan.yaml

```diff
@@ -0,0 +1,135 @@
+metadata:
+  name: codeqlExecuteScan
+  description: This step executes a codeql scan on the specified project to perform static code analysis and check the source code for security flaws.
+  longDescription: |-
+    This step executes a codeql scan on the specified project to perform static code analysis and check the source code for security flaws.
+
+    The codeql step triggers a scan locally on your orchestrator (e.g. Jenkins) within a docker container so finally you have to supply a docker image with codeql
+    and Java plus Maven.
+
+spec:
+  inputs:
+    secrets:
+      - name: githubTokenCredentialsId
+        description: Jenkins 'Secret text' credentials ID containing token to authenticate to GitHub.
+        type: jenkins
+    params:
+      - name: githubToken
+        description: "GitHub personal access token as per
+          https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line"
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        secret: true
+        aliases:
+          - name: access_token
+        resourceRef:
+          - name: githubTokenCredentialsId
+            type: secret
+          - type: vaultSecret
+            default: github
+            name: githubVaultSecretName
+      - name: buildTool
+        type: string
+        description: Defines the build tool which is used for building the project.
+        longDescription: |-
+          Based on the build tool the step will try to auto build the project. The step will try to auto select
+          the language and the build command. You can override the language and the build command by specifiying it seperatly.
+        mandatory: true
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        possibleValues:
+          - custom
+          - maven
+          - golang
+          - npm
+          - pip
+          - yarn
+        default: "maven"
+      - name: buildCommand
+        type: string
+        description: "Command to build the project"
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: language
+        type: string
+        description: "The programming language used to analyze."
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: modulePath
+        type: string
+        description: "Allows providing the path for the module to scan"
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: "./"
+      - name: codeqlQuery
+        type: string
+        description: "The name of a CodeQL query pack. If omitted, the default query suite for the language of the database being analyzed will be used."
```

Comment:
We should use the official terminology "query suite" both in the name of the parameter and in the description(s).

Reply:
Ok, I will change the query suite name.

Reply:
@vlkl-sap to give more info githubUrl(like github.com), commitId, ref(like refs/heads/main), and repository(like apache/camel) are mandatory to upload SARIF. The logic tries to auto-populate these, if not user has to pass them.